python/src/scipp/plot/plot_2d.py

```diff
@@ -282,20 +339,84 @@ def update_slice(self, change):
-                vslice = vslice[val.dim, val.value]
+                dslice = dslice[val.dim, val.value]
                 # At this point, after masks were combined, all their
                 # dimensions should be contained in the data_array.dims.
                 if self.params["masks"][self.name]["show"]:
                     mslice = mslice[val.dim, val.value]
             else:
-                button_dims[self.buttons[dim].value.lower() == "x"] = val.dim
+                # Get the dimensions of the dimension-coordinates, since
+                # buttons can contain non-dimension coordinates
+                button_dims[self.buttons[dim].value.lower() ==
+                            "x"] = self.slider_x[self.name][val.dim].dims[0]
 
         # Check if dimensions of arrays agree, if not, plot the transpose
-        slice_dims = vslice.dims
-        transp = slice_dims != button_dims
+        transp = dslice.dims != button_dims
+
+        # In the case of unaligned data, we may want to auto-scale the colorbar
+        # as we slice through dimensions. Colorbar limits are allowed to grow
+        # but not shrink.
+        autoscale_cbar = False
+        if dslice.unaligned is not None:
+            dslice = sc.histogram(dslice)
+            autoscale_cbar = True
+
+        # Make a new slice with bin edges and counts (for rebin), and with
+        # non-dimension coordinates if requested.
+        xy = "xy"
+        vslice = sc.DataArray(coords={
+            self.xyedges["x"].dims[0]: self.xyedges["x"],
+            self.xyedges["y"].dims[0]: self.xyedges["y"]
+        },
+                              data=sc.Variable(dims=[
+                                  self.xyedges[xy[not transp]].dims[0],
+                                  self.xyedges[xy[transp]].dims[0]
+                              ],
+                                               values=dslice.values,
+                                               variances=dslice.variances,
+                                               unit=sc.units.counts))
 
+        # Also include the masks
         if self.params["masks"][self.name]["show"]:
-            shape_list = [self.shapes[self.name][bdim] for bdim in button_dims]
+            mslice_dims = []
+            for dim in mslice.dims:
+                if dim == button_dims[0]:
+                    mslice_dims.append(self.xyedges["y"].dims[0])
+                elif dim == button_dims[1]:
+                    mslice_dims.append(self.xyedges["x"].dims[0])
+                else:
+                    mslice_dims.append(dim)
+            vslice.masks["all"] = sc.Variable(dims=mslice_dims,
+                                              values=mslice.values)
+
+        # The scaling by bin width and rebin operations below modify the
+        # variances in the data, so here we have to manually split the values
+        # and variances into separate Variables. We store them in a dict.
```

Comment:
Why does this imply that we have to split them? Do you need different error handling?

Reply:
Yes i need different error handling, because when i do https://github.com/scipp/scipp/blob/non_regular_bins_2d_plot/python/src/scipp/plot/plot_2d.py#L413, the errors grow in quadrature

Reply:
But that is how it is supposed to be, no?

Reply:
it is, but here i am trying to trick `rebin` into doing something it is not supposed to. i'm resampling an image just for the purposes of displaying it. if i don't do the split, the variances end up looking like this
![Screenshot_2020-06-12_14-31-26](https://user-images.githubusercontent.com/39047984/84502899-78924300-acb9-11ea-94a3-4b177efff3ba.png)
when they should actually be like this
![Screenshot_2020-06-12_14-31-36](https://user-images.githubusercontent.com/39047984/84502914-7fb95100-acb9-11ea-99e1-a5fc476275f7.png)

Reply:
That may be, but how do we know which one is correct? We need to follow through the logic of scaling by bin width, and how that is scaling the variances. We can look it up in Mantid, it is implemented there correctly (for example somewhere in `HistogramData/`.

Reply:
Mantid conversion from counts to frequencies (for variances): https://github.com/mantidproject/mantid/blob/7a904dff01c3b0dfef00ccea8d55f7b0715a077b/Framework/HistogramData/src/FrequencyVariances.cpp#L41.

Division by factor without variance used for division by bin width: https://github.com/scipp/scipp/blob/dde207e7d5ecab159c9dc27ed3d4fd6089cf27d2/core/include/scipp/core/value_and_variance.h#L140-L143.

I would conclude that the behavior should be correct if you just do the division by bin width, then rebin, then multiply by new width.

Reply:
After discussion:

What happens the other way round, when you are down-sampling? This is not clear what the variance be in an image pixel whichi is made up from several array elements in the underlying data.

It seems that if we cannot come up with a rule that works the same for both upsampling and downsampling, there probably is a flaw somewhere in your design.

Also, we are not sure how useful it is to show the variances, and it complicates a lot of the plotting code.

The decision is thus to remove showing the variances for 2d plots. If a user wants to look at the variances, they will either show up as error bars in 1d plots, or they can extract the variances and place them as values in a new Variable